spacy/tests/test_cli.py

```diff
@@ -750,6 +751,28 @@ def test_debug_data_compile_gold():
     assert data["boundary_cross_ents"] == 1
 
 
+@pytest.mark.issue(10727)
+def test_init_config_from_python_without_optional_args():
```

Comment:
I'm not sure we need this test? It ensures that an `AttributeError` is thrown when `init_config_cli` is called, but this isn't really the behaviour we're targeting, it's more like an unfortunate side effect we can work around by calling the other internal methods that now do have defaults? So maybe we should check the latter instead?

Reply:
True, that's a leftover. What would we check for `init_config()` though? Seems to me like the final change of adding default params is so trivial that we can forego the new test completely and rely on the existing ones.